src/main/java/com/minecolonies/coremod/entity/ai/mobs/EntityAIWalkToRandomHuts.java

```diff
@@ -0,0 +1,162 @@
+package com.minecolonies.coremod.entity.ai.mobs;
+
+import com.minecolonies.api.util.Log;
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.colony.ColonyManager;
+import com.minecolonies.coremod.colony.buildings.AbstractBuilding;
+import com.minecolonies.coremod.entity.pathfinding.GeneralEntityWalkToProxy;
+import com.minecolonies.coremod.entity.pathfinding.PathNavigate;
+import net.minecraft.entity.EntityCreature;
+import net.minecraft.entity.EntityLiving;
+import net.minecraft.entity.ai.EntityAIBase;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nullable;
+import java.lang.reflect.Field;
+import java.util.Collection;
+import java.util.Random;
+
+/**
+ * Created by Asher on 7/6/17.
+ */
+public class EntityAIWalkToRandomHuts extends EntityAIBase
+{
+
+    protected final EntityCreature entity;
+    private         BlockPos       targetBlock;
+    protected final World          world;
+    protected final double         speed;
+    protected final Colony         colony;
+
+    /**
+     * Walk to proxy.
+     */
+    private GeneralEntityWalkToProxy proxy;
+
+    /**
+     * The navigator for this entity.
+     */
+    private final PathNavigate newNavigator;
+
+    private Field navigatorField;
+
+    public EntityAIWalkToRandomHuts(final EntityCreature creatureIn, final double speedIn)
+    {
+        super();
+        this.entity = creatureIn;
+        this.speed = speedIn;
+        this.world = creatureIn.getEntityWorld();
+        this.colony = ColonyManager.getClosestColony(world, creatureIn.getPosition());
+        this.newNavigator = new PathNavigate(entity, world);
+        updateNavigatorField();
+        this.newNavigator.setCanSwim(true);
+        this.newNavigator.setEnterDoors(false);
+        this.setMutexBits(1);
+    }
+
+    @Override
+    public boolean shouldExecute()
+    {
+        if (this.targetBlock == null)
+        {
+            this.targetBlock = getRandomBuilding();
+        }
+
+        return this.targetBlock != null;
+    }
+
+    private synchronized void updateNavigatorField()
```

Comment:
Honestly, I'm not sure, all I know is that @Raycoms did quite a bit of the pathing, so I'm not sure (it's also mostly the same as the citizens pathing)